plugins/Carbonix/UI/TakeoffTab.cs

```diff
@@ -452,5 +461,46 @@ private void set_safety_switch(bool enable)
             var mode = new MAVLink.mavlink_set_mode_t() { custom_mode = custom_mode, target_system = (byte)Host.comPort.sysidcurrent };
             MainV2.comPort.setMode(mode, MAVLink.MAV_MODE_FLAG.SAFETY_ARMED);
         }
+
+        //reboot the flight conroller
+        private void but_reboot_Click(object sender, EventArgs e)
+        {
+            if (CustomMessageBox.Show("Are you sure you want to do " + but_reboot.Text + " ?", "Action",
+        MessageBoxButtons.YesNo) == (int)DialogResult.Yes)
+            {
+                try
+                {
+                    // Disable the button while we send the message
+                    ((Control)sender).Enabled = false;
+                    MainV2.comPort.doReboot();
+                }
+                catch
+                {
+                    CustomMessageBox.Show(Strings.CommandFailed, Strings.ERROR);
+                }
+            }
+        }
+
+        //One button to start and stop the engine. 
+        private void but_engControl_Click(object sender, EventArgs e)
+        {
+            if (engineRunning)
+            {
+                SendEngineCommand(false);
+                engineRunning = false;
+                but_engControl.Text = "Start Engine";
+            }
+            else
+            {
+                SendEngineCommand(true);
+                engineRunning= true;
+                but_engControl.Text = "Stop Engine";
+            }
+        }

```

Comment:
I think this should have a confirmation pop-up. At least for starting. I'd argue to have it for stopping too, but I won't fight if you disagree.

Don't really need the confirmation on the Emergency tab though.